components/ntp_background_images/browser/sponsored_images_component_data.h

```diff
@@ -13,9 +13,9 @@
 namespace ntp_background_images {
 
 struct SponsoredImagesComponentData {
-  std::string region;
-  std::string component_base64_public_key;
-  std::string component_id;
+  const char* region;
+  const char* component_base64_public_key;
+  const char* component_id;
```

Comment:
we need to be careful with such changes and do a double check. It might be breaking in such case:
```
bool CheckSomething(const SponsoredImagesComponentData& data, const char* input) {
  return data.region == input;
}
```

i.e. previously the check was running `std::string::operator==`, but now it may start to compare just two pointers.

I think we never use `const char*` as input parameters, but somewhere in the function logic it may appear. Just be aware.

Reply:
yes, agreed, I think I will change them to base::StringPiece